bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java

```diff
@@ -355,12 +355,16 @@ private BArray retrieveArrayValues(TomlNode tomlValue, String variableName, Arra
     private BMap<BString, Object> retrieveRecordValues(TomlNode tomlNode, String variableName, Type type) {
         RecordType recordType;
         String recordName;
+        Module recordPkg;
         if (type.getTag() == TypeTags.RECORD_TYPE_TAG) {
             recordName = type.getName();
             recordType = (RecordType) type;
+            recordPkg = recordType.getPackage();;
         } else {
-            recordName = ((BIntersectionType) type).getConstituentTypes().get(0).getName();
+            Type originalType = ((BIntersectionType) type).getConstituentTypes().get(0);
```

Comment:
Can non-readonly part come as `.get(1)` ?

Reply:
no. when I check with front-end we always have an order for intersectionType
https://github.com/ballerina-platform/ballerina-lang/blob/930969f1e3a43fff8a5ec2f74d8484b7df4bd0e9/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/ImmutableTypeCloner.java#L820